File: frontend/lib/models/habit_log.dart
class HabitLog {
  final String id;
  final String date;
  final bool completed;

  HabitLog({
    required this.id,
    required this.date,
    required this.completed,
  });

  factory HabitLog.fromJson(Map<String, dynamic> json) {
    return HabitLog(
      id: json['_id'],
      date: json['date'],
      completed: json['completed'],
    );
  }
}

File: backend/models/HabitLog.js
const mongoose = require("mongoose");

const habitLogSchema = new mongoose.Schema(
  {
    habitId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Habit",
      required: true,
    },

    date: {
      type: String, // YYYY-MM-DD
      required: true,
    },

    completed: {
      type: Boolean,
      default: false,
    },
  },
  {
    timestamps: true,
  }
);

// Prevent duplicate logs for same habit + date
habitLogSchema.index({ habitId: 1, date: 1 }, { unique: true });

module.exports = mongoose.model("HabitLog", habitLogSchema);
